torch/package/package_importer.py

```diff
@@ -337,20 +393,56 @@ def _make_module(
 
         return module
 
+    def _check_if_viable_parent(self, cur, child, name):
+        if isinstance(cur, _SelectiveExternNode):
+            return
+        if not isinstance(cur, (_PackageNode)) or child not in cur.children:
+            raise ModuleNotFoundError(
+                f'No module named "{name}" with in self-contained archive "{self.filename}"'
+                f" and the module is also not in the list of allowed external modules: {self.extern_modules}",
+                name=name,
+            )
+
+    def _get_node_from_name(self, name: str):
+        cur: _PathNode = self.root
+        atoms = name.split(".")
+        for atom in atoms[:-1]:
+            if isinstance(cur, _ExternNode):
+                return cur
+            if isinstance(cur, (_SelectiveExternNode, _PackageNode)):
+                self._check_if_viable_parent(cur, atom, name)
```

Comment:
Isn't it redundant to validate `name` here, as well as in `_load_module`? We do not mutate the module tree, so the result should be identical, right? If we are doing something slightly different than what was done in `_load_module`, please add a comment explaining what the difference is and why we need to do validation again.

Reply:
Yeah you're right it ends up being redundant